docs/installation.md

```diff
@@ -0,0 +1,365 @@
+# Installation
+
+This document provides detailed instructions on how to install the project. Follow these steps to set up the project on your local system.
+
+Instructions for Windows should theoretically work but we have only tested this project to work on -nix systems. As such, we are unable to verify any discrepancies on Windows, and there could be slight variations in your setup.
+
+## Prerequisites
+
+[Homebrew](https://brew.sh) is the simplest way to install system dependencies needed for this project.
+
+[Chocolatey](https://community.chocolatey.org/) is a good equivalent to Homebrew for package management in Windows.
+
+## System Requirements
+
+
+- **Node/NPM** (recommend using nvm)
+- **MariaDB** (using MariaDB in docker is also good)
+- **Java** (for Liquibase / OpenAPI  Generator)
+- **Liquibase**
+- **OpenAPI Generator**
+- **pyenv**
+- **wget** *(optional)*
+
+### Mac
+
+```bash
+brew install nvm
```

Comment:
Maybe not relevant for here, but I didn't realise how much `nvm` slows down the zshrc (like 80% of my startup cost), would it be worth recommending a lazy nvm loading approach, eg: https://github.com/undg/zsh-nvm-lazy-load/blob/master/zsh-nvm-lazy-load.plugin.zsh

Reply:
I almost want to just include `pnpm` into the instructions as a replacement for `npm`. Maybe we include `pnpm` instructions and also have a subsection for `nvm` if anyone wishes to stick to that. In that subsection I will include the lazy nvm approach. Thoughts?